src/Access/Authentication.h

```diff
@@ -103,7 +104,11 @@ class Authentication
 
     Type type = Type::NO_PASSWORD;
     Digest password_hash;
+
+    // Used and maintained only for LDAP.
     String server_name;
+    mutable std::size_t ldap_last_successful_password_check_params_hash = 0;
+    mutable std::chrono::steady_clock::time_point ldap_last_successful_password_check_timestamp;
```

Comment:
The same `User` instance can be used by multiple threads at the same time, and the function `Authentication::isCorrectPassword()` can be called for the same instance of `Authentication` (imagine someone are trying to connect to the server using the same user name). So it seems you have to think about some synchronization here, just adding `mutable` doesn't make the code correct.

Reply:
Good point, thanks. I am thinking about moving these fields (+ a mutex) from `Authentication` to a separate new member of `User` class called something like `cache`.

Reply:
Pushed a fix.